src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -50,6 +51,17 @@ export class ProsemirrorAdapter {
     @Event()
     private change: EventEmitter<string>;
 
+    @Watch('value')
+    public watchValue(newValue: string) {
+        if (newValue === '') {
```

Comment:
I think we should handle not this specific case but all the cases when the consumer might want to change the value of the component. It can be loading in a new template or do other things programatically in the consumer that changes the content of the text editor outside of prosemirror.